python-avd/pyavd/api/interface_descriptions/__init__.py

```diff
@@ -602,6 +602,10 @@ def __init__(  # noqa: PLR0913
         self.wan_circuit_id = wan_circuit_id
         self.main_interface_wan_carrier = main_interface_wan_carrier
 
+    @property
+    def hostname(self) -> str | None:
```

Comment:
please update if any of our methods use this today, so they pick it up from here instead.